src/main/java/co/cask/hydrator/plugin/spark/KinesisStreamingSource.java

```diff
@@ -0,0 +1,314 @@
+      if (recordFormat == null) {
+        outputSchema = config.parseSchema();
+        Schema messageSchema = config.parseSchema();
+        FormatSpecification spec =
+          new FormatSpecification(config.getFormat(), messageSchema, new HashMap<String, String>());
+        recordFormat = RecordFormats.createInitializedFormat(spec);
+      }
+      StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+      StructuredRecord messageRecord = recordFormat.read(new StreamEvent(ByteBuffer.wrap(data)));
+      for (Schema.Field messageField : messageRecord.getSchema().getFields()) {
+        String fieldName = messageField.getName();
+        builder.set(fieldName, messageRecord.get(fieldName));
+      }
+      return builder.build();
+    }
+  }
+
+  /**
+   * config file for Kinesis stream sink
+   */
+  public static class KinesisStreamConfig extends ReferencePluginConfig implements Serializable {
+
+    @Name("appName")
+    @Description(" The application name that will be used to checkpoint the Kinesis sequence numbers in DynamoDB table")
+    @Macro
+    private final String appName;
+
+    @Name("streamName")
+    @Description("The name of the Kinesis stream to the get the data from. The stream should be active")
+    @Macro
+    private final String streamName;
+
+    @Name("endpointUrl")
+    @Description("Valid Kinesis endpoints URL eg. kinesis.us-east-1.amazonaws.com")
+    @Macro
+    private final String endpoint;
+
+    @Name("region")
+    @Description("AWS region specific to the stream")
+    @Macro
+    private final String region;
+
+    @Name("duration")
+    @Description("The interval in milliseconds (e.g., Duration(2000) = 2 seconds) at which the Kinesis Client " +
+      "Library saves its position in the stream.")
+    @Macro
+    private final Integer duration;
+
+    @Name("initialPosition")
+    @Description("Can be either TRIM_HORIZON or LATEST, Default position will be Latest")
+    private final String initialPosition;
+
+    @Name("awsAccessKeyId")
+    @Description("AWS access Id having access to Kinesis streams")
+    @Macro
+    private final String awsAccessKeyId;
+
+    @Name("awsAccessSecret")
+    @Description("AWS access key secret having access to Kinesis streams")
+    @Macro
+    private final String awsAccessSecret;
+
+    @Description("Optional format of the Kinesis stream shard. Any format supported by CDAP is supported. For " +
+      "example, a value of 'csv' will attempt to parse Kinesis payloads as comma-separated values. If no format is " +
+      "given, Kinesis shard payloads will be treated as bytes.")
+    @Nullable
+    private final String format;
+
+    @Description("Output schema of the source, The fields are used in conjunction with the format to parse Kinesis " +
+      "payloads")
+    private final String schema;
+
+    public KinesisStreamConfig(String referenceName, String appName, String streamName, String endpoint, String region,
```

Comment:
The only optional field is format which is being handles in the gerFormat() function, thats why non parametrized constructor will not be required. Please advise.